spec/puppet/resource_api_spec.rb

```diff
@@ -1162,22 +1161,25 @@ def set(_context, changes)
 
           context 'with no changes' do
             it('set will not be called') do
+              instance.flush
+
               expect(instance.my_provider.last_changes).to be_nil
               expect(log_sink.last.message).to eq('Current State: {:name=>"somename", :test_string=>"canonfoo"}')
             end
           end
 
           context 'with a change' do
-            let(:test_string) { 'bar' }
-
             it('set will be called with the correct structure') do
-              expect(instance.my_provider.last_changes).to eq('somename' => {
-                                                                is: { name: 'somename', test_string: 'canonfoo' },
-                                                                should: { name: 'somename', test_string: 'canonbar' },
-                                                              })
-            end
+              run_one = type.new(name: 'somename', test_string: 'foo')
+              run_one.flush
+              run_two = type.new(name: 'somename', test_string: 'bar')
+              run_two.flush
```

Comment:
shouldn't this go into a `before` block, instead of conflating the two examples?